src/main/java/net/dv8tion/jda/core/requests/WebSocketClient.java

```diff
@@ -1006,32 +1008,43 @@ protected void handleEvent(JSONObject raw)
         }
     }
 
+    public boolean onBufferMessage(byte[] binary) throws IOException
```

Comment:
This should be private.

Reply:
It should probably also be synchronized: Either on the entire method or on a semaphore controlling access to `readBuffer`

Reply:
Why should this be synchronized? This only called on one thread